### ISSUE TITLE
Service Escalations do not send out Escalated recovery email notifications.

### ISSUE THREAD

**Issue by tgriep** [NagiosEnterprises/nagioscore#582]
Service Escalations in Core version 4.4.2 do not generate the recovery notification after the service returns to an OK stake.
They do send out the the Critical Notifications when the service is down but no notifications when the service recovers.

**Comment by wallabyies** [comment]
I just wanted to emphasize that not receiving recovery/OK notifications for escalated service alerts is a big problem for an operations team.       I hope this can be fixed soon, thanks!

**Comment by dvoryanchikov** [comment]
Hi Team!

**Debug**
The problem is that the recovery notification number is always set to 0 and thus recovery notifications never escalated.
 I believe this is because this line coming from d2f396d in fix of #550


**Workaround**
```
--- base/checks.c
+++ base/checks.c
@@ -894,7 +894,6 @@
 			svc->last_problem_id = svc->current_problem_id;
 			svc->current_problem_id = 0L;
             svc->current_attempt = 1;
-            svc->current_notification_number = 0;
             svc->host_problem_at_last_check = FALSE;
 		}
 
```

I've removed this line in maint copy and escalated recovery notifications are going as described in manual.
@scottwilkerson please take a look - I'm not sure this line can be simply removed not breaking anything...



**How to reproduce**
Examle config to reproduce this bug:

```

define timeperiod{
        timeperiod_name 24x7
        alias           24 Hours A Day, 7 Days A Week
        sunday          00:00-24:00
        monday          00:00-24:00
        tuesday         00:00-24:00
        wednesday       00:00-24:00
        thursday        00:00-24:00
        friday          00:00-24:00
        saturday        00:00-24:00
        }

define contact{
        contact_name                    duty
        host_notifications_enabled      0
        service_notifications_enabled   1
        host_notification_period        24x7
        service_notification_period     24x7
        host_notification_options       n
        service_notification_options    c,r
        host_notification_commands      dont-notify
        service_notification_commands   test-notify
}

define contact{
        contact_name                    boss
        host_notifications_enabled      0
        service_notifications_enabled   1
        host_notification_period        24x7
        service_notification_period     24x7
        host_notification_options       n
        service_notification_options    c,r
        host_notification_commands      dont-notify
        service_notification_commands   test-notify
}

define command{
        command_name    dont-notify
        command_line    /bin/true >/dev/null
        }

define command{
        command_name    test-notify
        command_line    /bin/echo "$SHORTDATETIME$ number $SERVICENOTIFICATIONNUMBER$ type $NOTIFICATIONTYPE$ recipients $NOTIFICATIONRECIPIENTS$ escalated $NOTIFICATIONISESCALATED$" >> /tmp/test_notifications.log
        }

define command{
        command_name    check_test
        command_line    $USER1$/check_dummy 0 "OK"
#	command_line	$USER1$/check_dummy 2 "KO"
}

define command{
        command_name    always_up
        command_line    $USER1$/check_dummy 0 "UP"
}

define host{
	host_name			somehost
	check_period			24x7
	notification_period             24x7
        max_check_attempts              2
        check_interval                  1
	retry_interval			1
        notification_interval           1
	check_command			always_up
	contacts			duty
	
}

define service{
        host_name                       somehost
        service_description             Test
	check_period			24x7
        notification_period             24x7
        max_check_attempts              2
        check_interval                  1
	retry_interval			1
        notification_interval           1
        check_command                   check_test
        contacts                        duty
        }

define serviceescalation{
        host_name                       somehost
        service_description             Test
        first_notification              4
        last_notification               0
        notification_interval           1
        contacts                        duty,boss
}


```

Tested by switching commented line in check_test and reloading.

**Log before patch**
```
08-11-2018 14:55:07 number 1 type PROBLEM recipients duty escalated 0
08-11-2018 14:58:07 number 2 type PROBLEM recipients duty escalated 0
08-11-2018 14:59:07 number 1 type RECOVERY recipients duty escalated 0
08-11-2018 15:06:07 number 1 type PROBLEM recipients duty escalated 0
08-11-2018 15:07:07 number 2 type PROBLEM recipients duty escalated 0
08-11-2018 15:08:07 number 3 type PROBLEM recipients duty escalated 0
08-11-2018 15:09:08 number 4 type PROBLEM recipients duty,boss escalated 1
08-11-2018 15:09:08 number 4 type PROBLEM recipients duty,boss escalated 1
08-11-2018 15:10:08 number 5 type PROBLEM recipients duty,boss escalated 1
08-11-2018 15:10:08 number 5 type PROBLEM recipients duty,boss escalated 1
08-11-2018 15:11:08 number 1 type RECOVERY recipients duty escalated 0
```

**Log after patch**
```
08-11-2018 16:43:17 number 1 type PROBLEM recipients duty escalated 0
08-11-2018 16:44:17 number 2 type PROBLEM recipients duty escalated 0
08-11-2018 16:45:17 number 3 type PROBLEM recipients duty escalated 0
08-11-2018 16:46:17 number 4 type PROBLEM recipients duty,boss escalated 1
08-11-2018 16:46:17 number 4 type PROBLEM recipients duty,boss escalated 1
08-11-2018 16:47:17 number 5 type PROBLEM recipients duty,boss escalated 1
08-11-2018 16:47:17 number 5 type PROBLEM recipients duty,boss escalated 1
08-11-2018 16:48:17 number 6 type RECOVERY recipients duty,boss escalated 1
08-11-2018 16:48:17 number 6 type RECOVERY recipients duty,boss escalated 1
08-11-2018 17:04:18 number 1 type PROBLEM recipients duty escalated 0
08-11-2018 17:05:18 number 2 type RECOVERY recipients duty escalated 0
08-11-2018 17:12:18 number 1 type PROBLEM recipients duty escalated 0
08-11-2018 17:13:18 number 2 type PROBLEM recipients duty escalated 0
08-11-2018 17:14:18 number 3 type PROBLEM recipients duty escalated 0
08-11-2018 17:15:18 number 4 type RECOVERY recipients duty escalated 0
```
Now it works as described in manual https://assets.nagios.com/downloads/nagioscore/docs/nagioscore/4/en/escalations.html
Will test if it reopens #550 later
__
BR,
Alexey

**Comment by dvoryanchikov** [comment]
>  Will test if it reopens #550 later
yes, this change unfortunately conflicts with #550, investigation needed..

**Comment by jomann09** [comment]
Hello @dvoryanchikov, did removing the changes that you mentioned above work?
If you're able to re-test the above issue with the current maint branch does it work? I actually moved the code that you tweaked (removed the single line from) to happen after notifications to fix another issue and I think it might also have fixed this issue. I will test it myself also.

**Comment by jomann09** [comment]
Okay I tested this with the latest `maint` branch and it seems to be working. Moving the code around to fix the soft state freezing also fixed this issue since notifications are handled before forced hard state is set.

Will be fixed in 4.4.3!

**Comment by wallabyies** [comment]
I confirmed this is resolved in Nagios XI 5.5.9.   Nice work!

**Comment by jomann09** [comment]
Awesome thanks for the response!
